File: (8 kyu) altERnaTIng cAsE = ALTerNAtiNG CaSe/(8 kyu) altERnaTIng cAsE = ALTerNAtiNG CaSe.cpp
#include <iostream>
#include <cctype>

std::string to_alternating_case(const std::string &str)
{
    std::string ret = str;
    for (char &c : ret)
    {
        c = isupper(c) ? tolower(c) : toupper(c);
    }
    return ret;
}

File: (8 kyu) altERnaTIng cAsE = ALTerNAtiNG CaSe/(8 kyu) altERnaTIng cAsE = ALTerNAtiNG CaSe.cs
using System;

namespace Extensions
{
  public static class StringExt
  {
    public static string ToAlternatingCase (this string s)
    {
      String res = String.Empty;
      
      for (int i = 0; i < s.Length; i++)
      {
          if (s[i].ToString() == s[i].ToString().ToUpper())
          {
              res += s[i].ToString().ToLower();
          }
          else
          {
              res += s[i].ToString().ToUpper();
          }
      }
      
      return res;
    }
  }
}
